File: core/src/main/java/com/shmedo/core/util/AndroidVersion.java
package com.shmedo.core.util;

import android.os.Build;

/**
 * 以更加可读的方式提供Android系统版本号的判断方法。
 *
 * @author：gonghe
 * @time 2019/4/9
 */
public class AndroidVersion {
    /**
     * 判断当前手机系统版本API是否是16以上。
     *
     * @return 16以上返回true，否则返回false。
     */
    public static boolean hasJellyBean() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN;
    }


    /**
     * 判断当前手机系统版本API是否是17以上。
     *
     * @return 17以上返回true，否则返回false。
     */
    public static boolean hasJellyBeanMR1() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1;
    }

    /**
     * 判断当前手机系统版本API是否是18以上。
     *
     * @return 18以上返回true，否则返回false。
     */
    public static boolean hasJellyBeanMR2() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR2;
    }

    /**
     * 判断当前手机系统版本API是否是19以上。
     *
     * @return 19以上返回true，否则返回false。
     */
    public static boolean hasKitkat() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.KITKAT;
    }


    /**
     * 判断当前手机系统版本API是否是21以上。
     *
     * @return 21以上返回true，否则返回false。
     */
    public static boolean hasLollipop() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP;
    }

    /**
     * 判断当前手机系统版本API是否是22以上。
     *
     * @return 22以上返回true，否则返回false。
     */
    public static boolean hasLollipopMR1() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP_MR1;
    }

    /**
     * 判断当前手机系统版本API是否是23以上。
     *
     * @return 23以上返回true，否则返回false。
     */
    public static boolean hasMarshmallow() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.M;
    }

    /**
     * 判断当前手机系统版本API是否是24以上。
     *
     * @return 24以上返回true，否则返回false。
     */
    public static boolean hasNougat() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.N;
    }

    /**
     * 判断当前手机系统版本API是否是26以上。
     *
     * @return 26以上返回true，否则返回false。
     */
    public static boolean hasOreo() {
        return Build.VERSION.SDK_INT >= Build.VERSION_CODES.O;
    }
}

File: core_commonlib/src/main/java/com/shmedo/core/commonlib/utils/AndroidVersion.kt
package com.shmedo.core.commonlib.utils

import android.annotation.SuppressLint
import android.content.Context
import android.os.Build

/**
 * author : Android 轮子哥
 * github : https://github.com/getActivity/XXPermissions
 * time   : 2022/01/22
 * desc   : Android 版本判断
 */
@SuppressLint("AnnotateVersionCheck")
object AndroidVersion {
//    val ANDROID_14: Int = Build.VERSION_CODES.UPSIDE_DOWN_CAKE
    const val ANDROID_13 = Build.VERSION_CODES.TIRAMISU
    const val ANDROID_12_L = Build.VERSION_CODES.S_V2
    const val ANDROID_12 = Build.VERSION_CODES.S
    const val ANDROID_11 = Build.VERSION_CODES.R
    const val ANDROID_10 = Build.VERSION_CODES.Q
    const val ANDROID_9 = Build.VERSION_CODES.P
    const val ANDROID_8_1 = Build.VERSION_CODES.O_MR1
    const val ANDROID_8 = Build.VERSION_CODES.O
    const val ANDROID_7_1 = Build.VERSION_CODES.N_MR1
    const val ANDROID_7 = Build.VERSION_CODES.N
    const val ANDROID_6 = Build.VERSION_CODES.M
    const val ANDROID_5_1 = Build.VERSION_CODES.LOLLIPOP_MR1
    const val ANDROID_5 = Build.VERSION_CODES.LOLLIPOP
    const val ANDROID_4_4 = Build.VERSION_CODES.KITKAT
    const val ANDROID_4_3 = Build.VERSION_CODES.JELLY_BEAN_MR2
    const val ANDROID_4_2 = Build.VERSION_CODES.JELLY_BEAN_MR1
    const val ANDROID_4_1 = Build.VERSION_CODES.JELLY_BEAN
    const val ANDROID_4_0 = Build.VERSION_CODES.ICE_CREAM_SANDWICH
    val androidVersionCode: Int
        /**
         * 获取 Android 版本码
         */
        get() = Build.VERSION.SDK_INT

    /**
     * 获取 targetSdk 版本码
     */
    fun getTargetSdkVersionCode(context: Context): Int {
        return context.applicationInfo.targetSdkVersion
    }

    val isAndroid14: Boolean
        /**
         * 是否是 Android 14 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= 34
    val isAndroid13: Boolean
        /**
         * 是否是 Android 13 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_13
    val isAndroid12: Boolean
        /**
         * 是否是 Android 12 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_12
    val isAndroid11: Boolean
        /**
         * 是否是 Android 11 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_11
    val isAndroid10: Boolean
        /**
         * 是否是 Android 10 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_10
    val isAndroid9: Boolean
        /**
         * 是否是 Android 9.0 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_9
    val isAndroid8: Boolean
        /**
         * 是否是 Android 8.0 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_8
    val isAndroid7_1: Boolean
        /**
         * 是否是 Android 7.1 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_7_1
    val isAndroid7: Boolean
        /**
         * 是否是 Android 7.0 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_7
    val isAndroid6: Boolean
        /**
         * 是否是 Android 6.0 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_6
    val isAndroid5_1: Boolean
        /**
         * 是否是 Android 5.1 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_5_1
    val isAndroid5: Boolean
        /**
         * 是否是 Android 5.0 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_5
    val isAndroid4_4: Boolean
        /**
         * 是否是 Android 4.4 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_4_4
    val isAndroid4_3: Boolean
        /**
         * 是否是 Android 4.3 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_4_3
    val isAndroid4_2: Boolean
        /**
         * 是否是 Android 4.2 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_4_2
    val isAndroid4: Boolean
        /**
         * 是否是 Android 4.0 及以上版本
         */
        get() = Build.VERSION.SDK_INT >= ANDROID_4_0
}